### PR TITLE
LPD-30357 - Refactor stylebook import/export classes

### PR DESCRIPTION
**Pull request by ethib137**
Resent from #784

# Original description
## Motivation
Currently, the service code responsible for import/export a stylebook was living in a web module, instead of a service module, which its not the proper place to be, since web module should interacts with service/impl modules through api.

## Proposed Solution
Move code to service module.

### CODE REVIEW

**Comment by ethib137** [comment]
@algobob & @thiago-buarqque 

For refactors we should be handling our commits similar to how I am in this PR.

- When files are being moved or renamed, that should be handled in 1 commit. Git is smart enough to still compare these files so it makes it easier to check that the file is really just being moved or renamed.
- No logic changes should happen in rename or refactor commits.
- Renames should be as consistent as possible. This is why I added a separate commit for [Rename function name to match the return type.](https://github.com/liferay-platform-experience/liferay-portal/pull/785/commits/8b0355cc645a594def891c9dd5d2b0c9883e721f).

FYI @algobob we had agreed upon using the name [StyleBookEntryImporterResult](https://github.com/liferay-platform-experience/liferay-portal/pull/783#discussion_r1831355310) but it wasn't included in the refactor. 

Please make sure I didn't make any mistakes or break anything. If it all looks good, feel free to send it to BPM. Make sure to let them know that it is your work and I just modified the commit history.

Thanks guys.

**Comment by ethib137** [comment]
ci:test:sf

**Comment by ethib137** [comment]
[This comment](https://github.com/liferay-core-infra/liferay-portal/pull/7606#discussion_r1831150240) is also really important to always be considering with your commits.

> Please make sure you reflect this in the commit message, so in the future when people look at this part they can also know the context.

Commit messages should always be used to explain why the changes you're making are necessary and why you're doing it the way you are doing it. Thanks!

**Comment by thiago-buarqque** [comment]
@ethib137 In the first commit you have renamed [_importResultEntries to _styleBookEntryImporterEntries](https://github.com/liferay-platform-experience/liferay-portal/pull/785/commits/2ed910e7a8d8a16212f169aa64b32ecef385fecd#diff-9685d8d9987ffdac96645ff57a8a77f9c355b3cc620346ae94dd51e1e3616e76L81-R58) to match the actual type of the List. 

In the [second commit](https://github.com/liferay-platform-experience/liferay-portal/pull/785/commits/dc14f387aee337384b47be16cf9245de17817ab1#diff-c47adb5ad7c5a32047d098027e012d034a563c33adb5701e5fdf6c4b036b4d42L50-R48) you renamed  _getStyleBookEntryZipProcessorImportResultEntryNames to _getStyleBookEntryImporterResultNames, keeping the wrong naming and then changing it in the third commit.

Why isn't the first renaming not in the second commit? And why isn't the third commit inside the second one as well?

**Comment by thiago-buarqque** [comment]
You didn't update all usages of `StyleBookEntryImporterEntry` to `StyleBookEntryImporterResult` in `stylebookentryimporterimpl.java`

**Comment by thiago-buarqque** [comment]
@ethib137 I do get your point, better to review, better to read, better to understand. Thanks for you insights, I will be looking carefully towards it from now on

File: modules/apps/style-book/style-book-web/src/main/java/com/liferay/style/book/web/internal/display/context/ImportStyleBookDisplayContext.java [hunk 1]
```diff
@@ -23,48 +23,42 @@ public ImportStyleBookDisplayContext(RenderRequest renderRequest) {
...
 		}
 
 		return TransformUtil.transform(
-			styleBookEntryZipProcessorImportResultEntries,
-			styleBookEntryZipProcessorImportResultEntry -> {
-				if (styleBookEntryZipProcessorImportResultEntry.getStatus() !=
-						status) {
-
+			styleBookEntryImporterResults,
+			styleBookEntryImporterEntry -> {
```
Comment (thiago-buarqque) [inline modules/apps/style-book/style-book-web/src/main/java/com/liferay/style/book/web/internal/display/context/ImportStyleBookDisplayContext.java:38]: Should be `styleBookEntryImporterResult`

**Comment by ethib137** [comment]
> @ethib137 In the first commit you have renamed [_importResultEntries to _styleBookEntryImporterEntries](https://github.com/liferay-platform-experience/liferay-portal/pull/785/commits/2ed910e7a8d8a16212f169aa64b32ecef385fecd#diff-9685d8d9987ffdac96645ff57a8a77f9c355b3cc620346ae94dd51e1e3616e76L81-R58) to match the actual type of the List.
> 
> In the [second commit](https://github.com/liferay-platform-experience/liferay-portal/pull/785/commits/dc14f387aee337384b47be16cf9245de17817ab1#diff-c47adb5ad7c5a32047d098027e012d034a563c33adb5701e5fdf6c4b036b4d42L50-R48) you renamed _getStyleBookEntryZipProcessorImportResultEntryNames to _getStyleBookEntryImporterResultNames, keeping the wrong naming and then changing it in the third commit.
> 
> Why isn't the first renaming not in the second commit? 

@thiago-buarqque yep I rushed this to try and give you guys an idea of what I'm looking for. @algobob can you redo the commits in this general way?

> And why isn't the third commit inside the second one as well?

This is a different type of rename since it's not just changing the name because the Class changed. We are changing this name because it had the wrong name for the return type. It's best to draw attention to why we are changing it, so that it's more explicit.

So any time there's a different reason for why you are refactoring something it should be in a different commit so that the commit message can explain the reason.

**Comment by algobob** [comment]
Stepping back from this PR for now to handle other story. I'll jump back after.

**Pull request closed without merging**